pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -15,14 +15,213 @@
+        in a completed execution with incorrect results.
+
+    Args:
+        wires (Wires): The wire to measure.
+        reset (Optional[bool]): Whether to reset the wire to the :math:`|0 \rangle`
+            state after measurement.
+        postselect (Optional[int]): Which basis state to postselect after a mid-circuit
+            measurement. None by default. If postselection is requested, only the post-measurement
+            state that is used for postselection will be considered in the remaining circuit.
+
+    Returns:
+        MeasurementValue: the mid-circuit measurement result linked to the created MidMeasureMP
+
+    Raises:
+        QuantumFunctionError: if multiple wires were specified
+
+    """
+
+    # ToDo: if capture is enabled, create and bind primitive here and return primitive instead (subsequent PR)
+
+    return _measure_impl(wires, XMidMeasureMP, reset=reset, postselect=postselect)
+
+
+def measure_y(
+    wires: Union[Hashable, Wires],
+    reset: bool = False,
+    postselect: Optional[int] = None,
+):
+    r"""Perform a mid-circuit measurement in the Y basis. The measurements are performed using the 0, 1
+    convention rather than the ±1 convention.
+
+    For more details on the results of mid-circuit measurements and how to use them,
+    see :func:`qml.measure <pennylane.measure>`.
+
+    For more details on mid-circuit measurements in an arbitrary basis (besides the computational basis),
+    see :func:`measure_arbitrary_basis <pennylane.ftqc.measure_arbitrary_basis>`.
+
+    .. warning::
+        Measurements should be diagonalized before execution for any device that only natively supports
+        mid-circuit measurements in the computational basis. To diagonalize, the :func:`diagonalize_mcms <pennylane.ftqc.diagonalize_mcms>`
+        transform can be applied.
+
+        Skipping diagonalization for a circuit containing parametric mid-circuit measurements may result
+        in a completed execution with incorrect results.
+
+    Args:
+        wires (Wires): The wire to measure.
+        reset (Optional[bool]): Whether to reset the wire to the :math:`|0 \rangle`
+            state after measurement.
+        postselect (Optional[int]): Which basis state to postselect after a mid-circuit
+            measurement. None by default. If postselection is requested, only the post-measurement
+            state that is used for postselection will be considered in the remaining circuit.
+
+    Returns:
+        MeasurementValue: the mid-circuit measurement result linked to the created MidMeasureMP
+
+    Raises:
+        QuantumFunctionError: if multiple wires were specified
+
+    """
+
+    # ToDo: if capture is enabled, create and bind primitive here and return primitive instead (subsequent PR)
+
+    return _measure_impl(wires, YMidMeasureMP, reset=reset, postselect=postselect)
+
+
+def _measure_impl(
+    wires: Union[Hashable, Wires],
+    measurement_class=MidMeasureMP,
+    **kwargs,
```

Comment:
The plan would be to also use this in capture for the `def_impl` method, as with the standard MCM - and it skips duplicating the wire checks and creating the id.  @mudit2812, do you have any concerns about having generic `**kwargs` as an argument (and could that be an issue with program capture)? We can split this up again a bit if we want to keep things more defined, and have `_measure_x_impl`, `_measure_y_impl`, etc.

Reply:
I think it should be fine, but since the changes are "leaking" out of the `ftqc` module, it might be better for now to leave `measure_prim` as is and create new ones for `measure_x` and `measure_y` in the `ftqc` module.

I'm wondering it if it make sense to make `MidMeasureMP` be a subclass of `ParametricMidMeasureMP` rather than the other way around. `qml.measure` can then be an alias for `measure_z`, and a unified `measure` primitive could have more clear kwargs (always `reset`, `postselect`, `angle`, and `plane`). That seems like a longer term change as moving stuff out of the `ftqc` module will require product considerations as well.

Reply:
Agreed, I think we'll need separate primitives for those, since we don't want to start changing things outside this module - and if we eventually have workflows where we don't want to decompose them, I think it will be simpler to have individual representation instead of storing everything as arbitrary basis MCMs. 

I'll try to keep things in a similar format to the main `mid_measure` implementation where it makes sense, so that unifying them in main PennyLane is easier if we want to do that in the future. At that point, switching things up to `ParametricMidMeasureMP` might be more organized, that's an interesting point.